gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
+    """
+    if scipy.sparse.issparse(vec1):
+        vec1 = numpy.asarray(vec1.todense())
+    if scipy.sparse.issparse(vec2):
+        vec2 = numpy.asarray(vec2.todense())
+    if isinstance(vec1, numpy.ndarray):
+        vec1 = vec1.tolist()
+    if isinstance(vec2, numpy.ndarray):
+        vec2 = vec2.tolist()
+    if isbow(vec1) and isbow(vec2):
+        max_len = max(len(vec1), len(vec2), num_features)
+        dense1 = sparse2full(vec1, max_len)
+        dense2 = sparse2full(vec2, max_len)
+        return scipy.stats.entropy(dense1, dense2)
+    else:
+        return scipy.stats.entropy(vec1, vec2)
+
+
+def hellinger(vec1, vec2):
+    """
+    Hellinger distance is a distance metric to quanitfy the similarity between two probability distributions.
+    Similarity between distributions will be a number between <0,1>, where 0 is maximum similarity and 1 is minimum similarity.
+    """
+    if scipy.sparse.issparse(vec1):
+        vec1 = numpy.asarray(vec1.todense())
+    if scipy.sparse.issparse(vec2):
+        vec2 = numpy.asarray(vec2.todense())
+    if isinstance(vec1, numpy.ndarray):
+        vec1 = vec1.tolist()
+    if isinstance(vec2, numpy.ndarray):
+        vec2 = vec2.tolist()
+    if isbow(vec1) and isbow(vec2):
+        vec1, vec2 = dict(vec1), dict(vec2)
+        if len(vec2) < len(vec1):
+            vec1, vec2 = vec2, vec1
+        sim = numpy.sqrt(0.5*sum((numpy.sqrt(value) - numpy.sqrt(vec2.get(index, 0.0)))**2 for index, value in iteritems(vec1)))
+        return sim
+    else:
+        sim = numpy.sqrt(0.5 * ((numpy.sqrt(vec1) - numpy.sqrt(vec2))**2).sum())
+        return sim
+
+def jaccard(vec1,vec2):
+    """
+    A similarity metric between bags of words representation.
+    Return the intersection divided by union, where union is the sum of the size of the two bags.
+    Highest value jaccard similarity of two bags is 1/2, which indicates the highest similarity.
+    If it is not a bag of words representation, the union and intersection is calculated in the traditional manner.
+    Returns a similarity in range <0,1> where values closer to 1 mean a higher similarity.
+    """
+
+    if isbow(vec1) and isbow(vec2):
+        if scipy.sparse.issparse(vec1):
+            vec1 = numpy.asarray(vec1.todense())
+        if scipy.sparse.issparse(vec2):
+            vec2 = numpy.asarray(vec2.todense())
+        intersection = 0
+        union = sum(weight for id_,weight in vec1) + sum(weight for id_,weight in vec2)
+        vec1, vec2 = dict(vec1), dict(vec2)
+        for item in vec1:
+            if item in vec2:
+                intersection = min(vec2[item],vec1[item]) + intersection
+        return(float(intersection)/float(union))
+    else:
+        if scipy.sparse.issparse(vec1):
+            vec1 = numpy.asarray(vec1.todense())
+        if scipy.sparse.issparse(vec2):
+            vec2 = numpy.asarray(vec2.todense())
+        if isinstance(vec1, numpy.ndarray) and len(vec1) == 1:
```

Comment:
If it is not a bow and an `ndarray` then it will look like this, for e.g - `numpy.array([[6, 1, 1, 9]])`, and to use it as a list later I need to get only `[6,1,1,9]`, so I check length == 1. If it is any other length and also not a BOW that input sequence can't be used for `jaccard`. 
The reason I convert it to a list is to check for intersection and union, but that seems redundant now considering `set` and `frozenset` like you mentioned below.